packages/backend/src/lib/backends/base.sync.ts

```diff
@@ -24,6 +24,33 @@ import { InsightSyncTask } from '../../models/tasks';
 import { ActivityService } from '../../services/activity.service';
 import { InsightService } from '../../services/insight.service';
 
+export const THUMBNAIL_EXTENSIONS = ['.png', '.jpg', '.jpeg', '.gif', '.svg'];
+export const THUMBNAIL_LOCATIONS = ['thumbnail', '.iex/thumbnail'].flatMap((prefix) =>
+  THUMBNAIL_EXTENSIONS.map((extension) => prefix + extension)
+);
+
+export const READONLY_FILES = new Set(['insight.yml']);
+
+export const INDEXABLE_MIME_TYPES = new Set([
+  'application/x-ipynb+json',
+  'application/javascript',
+  'application/json',
+  'application/x-sh',
+  'application/x-sql',
+  'application/x-typescript',
+  'application/xml',
+  'text/html',
+  'text/markdown',
+  'text/plain',
+  'text/x-clojure',
+  'text/x-groovy',
+  'text/x-java-source',
+  'text/x-python',
+  'text/x-ruby',
+  'text/x-scala',
+  'text/yaml'
+]);
+
```

Comment:
This section is just refactored out of `github.sync.ts` since it is shared between different sync backends.